dnf/conf/substitutions.py

```diff
@@ -37,6 +37,8 @@ def _update_from_env(self):
         numericvars = ['DNF%d' % num for num in range(0, 10)]
         for key, val in os.environ.items():
             if ENVIRONMENT_VARS_RE.match(key) or key in numericvars:
+                self[key.lstrip("DNF_VAR_")] = val
```

Comment:
This is wrong - the argument of lstrip is not a substring, but set of characters to strip! So now it also strips DNF0 -> 0

Reply:
I am sorry, I forget to remove `or key in numericvars` from `if`.